processor/otel/consumer.go

```diff
@@ -296,12 +296,31 @@ func translateTransaction(
 			case conventions.AttributeNetHostName:
 				netHostName = stringval
 
-			// messaging
-			//
-			// TODO(axw) translate OpenTelemtry messaging conventions.
-			case "message_bus.destination":
+			// messaging.*
+			case "message_bus.destination", conventions.AttributeMessagingDestination:
 				message.QueueName = stringval
-				isMessaging = true
+				isMessagingSpan = true
+			case "messaging.rabbitmq.routing_key":
+				message.RoutingKey = stringval
+				isMessagingSpan = true
+			case conventions.AttributeMessagingOperation:
+				if stringval == "" {
+					// TODO: Is this correct? Inferring that an empty string means "send", based on
+					// only `send`, `process`, and `receive` are allowed
+					// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#operation-names
+					// `messaging.operation` can only have `process` and `receive` values
+					// cf. table in section https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#apache-kafka-example
+					// and various examples
+					// https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#apache-kafka-example
+					stringval = "send"
+				}
+				message.Operation = stringval
+				isMessagingSpan = true
+			case conventions.AttributeMessagingSystem:
+				message.System = stringval
+				destinationService.Resource = stringval
+				destinationService.Name = stringval
```

Comment:
does this compile? transactions shouldn't have a destination service, only spans should

Reply:
Ah, I cargo-culted that from the span code and forgot to check if it recompiled (it doesn't, and all of that code has been removed now anyway)